code/lang/ILanguage.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using Sandbox;
+
+namespace TTTReborn.Language
+{
+    public interface ILanguage
+    {
+        // TODO: On cvar change callback
+        [ClientVar("ttt_language", Help = "Your current language", Saved = true)]
+        public static string ActiveLanguage { get; set; }
+
+        [ServerVar("ttt_server_language", Help = "Server's current language", Saved = true)]
+        public static string ServerLanguage { get; set; }
+        public const string FALLBACK_LANGUAGE = "English";
```

Comment:
I'd like to avoid these language names and instead using "enUS" or other ISO styles to support region based language specifications as well :)

Maybe we should add a json header / config for that language as well then (original language name)? Then we should keep the outer brackets of the json for sure.

Reply:
I thought about that. But in this case I had to rewrite TLanguage constructor. :D

Reply:
Languages could be specified by an enum with a single invariant culture language (English) that specifies keys for localized string lookups.

Reply:
> Languages could be specified by an enum with a single invariant culture language (English) that specifies keys for localized string lookups.

That's not a good idea as language files should be loaded on fly instead of needing to interact with the API / C# code directly.

Maybe we should create a default folder structure, so all files inside a lang folder that are json files are getting initialized. And that eg. a enUS.json or config.json contains the relevant data. Or just combined inside one file via json syntax (just reserve they keyword "config" e.g.)